# Carga de datos con Pandas
En esta actividad aprenderás a leer un archivo de datos csv utilizando la librería Pandas.

**1.** Importa pandas.

In [8]:
import pandas as pd

**2.** Lee el archivo .csv usando la función read_csv.

In [9]:
avocado = pd.read_csv('avocado.csv')

In [10]:
avocado

Unnamed: 0        Date  AveragePrice  Total Volume     4046       4225  \
0               0  2015-12-27          1.33      64236.62  1036.74   54454.85   
1               1  2015-12-20          1.35      54876.98   674.28   44638.81   
2               2  2015-12-13          0.93     118220.22   794.70  109149.67   
3               3  2015-12-06          1.08      78992.15  1132.00   71976.41   
4               4  2015-11-29          1.28      51039.60   941.48   43838.39   
...           ...         ...           ...           ...      ...        ...   
18244           7  2018-02-04          1.63      17074.83  2046.96    1529.20   
18245           8  2018-01-28          1.71      13888.04  1191.70    3431.50   
18246           9  2018-01-21          1.87      13766.76  1191.92    2452.79   
18247          10  2018-01-14          1.93      16205.22  1527.63    2981.04   
18248          11  2018-01-07          1.62      17489.58  2894.77    2356.13   

         4770  Total Bags  Small Bags  Large Bags  XLarge Bags          type  \
0       48.16     8696.87     8603.62       93.25          0.0  conventional   
1       58.33     9505.56     9408.07       97.49          0.0  conventional   
2      130.50     8145.35     8042.21      103.14          0.0  conventional   
3       72.58     5811.16     5677.40      133.76          0.0  conventional   
4       75.78     6183.95     5986.26      197.69          0.0  conventional   
...       ...         ...         ...         ...          ...           ...   
18244    0.00    13498.67    13066.82      431.85          0.0       organic   
18245    0.00     9264.84     8940.04      324.80          0.0       organic   
18246  727.94     9394.11     9351.80       42.31          0.0       organic   
18247  727.01    10969.54    10919.54       50.00          0.0       organic   
18248  224.53    12014.15    11988.14       26.01          0.0       organic   

       year            region  
0      2015            Albany  
1      2015            Albany  
2      2015            Albany  
3      2015            Albany  
4      2015            Albany  
...     ...               ...  
18244  2018  WestTexNewMexico  
18245  2018  WestTexNewMexico  
18246  2018  WestTexNewMexico  
18247  2018  WestTexNewMexico  
18248  2018  WestTexNewMexico  

[18249 rows x 14 columns]

**3.** Observa las primeras líneas usando la función head.

In [19]:
avocado.head()

Unnamed: 0        Date  AveragePrice  Total Volume     4046       4225  \
0           0  2015-12-27          1.33      64236.62  1036.74   54454.85   
1           1  2015-12-20          1.35      54876.98   674.28   44638.81   
2           2  2015-12-13          0.93     118220.22   794.70  109149.67   
3           3  2015-12-06          1.08      78992.15  1132.00   71976.41   
4           4  2015-11-29          1.28      51039.60   941.48   43838.39   

     4770  Total Bags  Small Bags  Large Bags  XLarge Bags          type  \
0   48.16     8696.87     8603.62       93.25          0.0  conventional   
1   58.33     9505.56     9408.07       97.49          0.0  conventional   
2  130.50     8145.35     8042.21      103.14          0.0  conventional   
3   72.58     5811.16     5677.40      133.76          0.0  conventional   
4   75.78     6183.95     5986.26      197.69          0.0  conventional   

   year  region  
0  2015  Albany  
1  2015  Albany  
2  2015  Albany  
3  2015  Albany  
4  2015  Albany

**4.** Obtén diferentes combinaciones de columnas y datos.

In [29]:
avocado.iloc[1:, 0:5]

Unnamed: 0        Date  AveragePrice  Total Volume     4046
1               1  2015-12-20          1.35      54876.98   674.28
2               2  2015-12-13          0.93     118220.22   794.70
3               3  2015-12-06          1.08      78992.15  1132.00
4               4  2015-11-29          1.28      51039.60   941.48
5               5  2015-11-22          1.26      55979.78  1184.27
...           ...         ...           ...           ...      ...
18244           7  2018-02-04          1.63      17074.83  2046.96
18245           8  2018-01-28          1.71      13888.04  1191.70
18246           9  2018-01-21          1.87      13766.76  1191.92
18247          10  2018-01-14          1.93      16205.22  1527.63
18248          11  2018-01-07          1.62      17489.58  2894.77

[18248 rows x 5 columns]

In [27]:
avocado.iloc[1:, [1,3,6,7,9]]

Date  Total Volume    4770  Total Bags  Large Bags
1      2015-12-20      54876.98   58.33     9505.56       97.49
2      2015-12-13     118220.22  130.50     8145.35      103.14
3      2015-12-06      78992.15   72.58     5811.16      133.76
4      2015-11-29      51039.60   75.78     6183.95      197.69
5      2015-11-22      55979.78   43.61     6683.91      127.44
...           ...           ...     ...         ...         ...
18244  2018-02-04      17074.83    0.00    13498.67      431.85
18245  2018-01-28      13888.04    0.00     9264.84      324.80
18246  2018-01-21      13766.76  727.94     9394.11       42.31
18247  2018-01-14      16205.22  727.01    10969.54       50.00
18248  2018-01-07      17489.58  224.53    12014.15       26.01

[18248 rows x 5 columns]

**5.** Revisa los datos obtenidos por la función  describe.

In [13]:
avocado.describe()

Unnamed: 0  AveragePrice  Total Volume          4046          4225  \
count  18249.000000  18249.000000  1.824900e+04  1.824900e+04  1.824900e+04   
mean      24.232232      1.405978  8.506440e+05  2.930084e+05  2.951546e+05   
std       15.481045      0.402677  3.453545e+06  1.264989e+06  1.204120e+06   
min        0.000000      0.440000  8.456000e+01  0.000000e+00  0.000000e+00   
25%       10.000000      1.100000  1.083858e+04  8.540700e+02  3.008780e+03   
50%       24.000000      1.370000  1.073768e+05  8.645300e+03  2.906102e+04   
75%       38.000000      1.660000  4.329623e+05  1.110202e+05  1.502069e+05   
max       52.000000      3.250000  6.250565e+07  2.274362e+07  2.047057e+07   

               4770    Total Bags    Small Bags    Large Bags    XLarge Bags  \
count  1.824900e+04  1.824900e+04  1.824900e+04  1.824900e+04   18249.000000   
mean   2.283974e+04  2.396392e+05  1.821947e+05  5.433809e+04    3106.426507   
std    1.074641e+05  9.862424e+05  7.461785e+05  2.439660e+05   17692.894652   
min    0.000000e+00  0.000000e+00  0.000000e+00  0.000000e+00       0.000000   
25%    0.000000e+00  5.088640e+03  2.849420e+03  1.274700e+02       0.000000   
50%    1.849900e+02  3.974383e+04  2.636282e+04  2.647710e+03       0.000000   
75%    6.243420e+03  1.107834e+05  8.333767e+04  2.202925e+04     132.500000   
max    2.546439e+06  1.937313e+07  1.338459e+07  5.719097e+06  551693.650000   

               year  
count  18249.000000  
mean    2016.147899  
std        0.939938  
min     2015.000000  
25%     2015.000000  
50%     2016.000000  
75%     2017.000000  
max     2018.000000

**6.** Responde las preguntas de la sección anterior.
* ¿Cuántos objetos hay?

    18248


In [33]:
avocado.count()

Unnamed: 0      18249
Date            18249
AveragePrice    18249
Total Volume    18249
4046            18249
4225            18249
4770            18249
Total Bags      18249
Small Bags      18249
Large Bags      18249
XLarge Bags     18249
type            18249
year            18249
region          18249
dtype: int64

* ¿Cuál es el valor de la variable 2 del objeto 15?

    1.28

In [34]:
avocado.loc[15,'AveragePrice']

1.28



* ¿Cuál es el mínimo y máximo de la variable 3?

    El máximo es 6.250565e+07
    
    El mínimo es 8.456000e+01